bundle/manifests/argoproj.io_argocds.yaml

```diff
@@ -1480,25 +1564,64 @@ spec:
                               the short service name (service.namespace.svc), which allows infrastructure generated certificates to automatically
                               verify.
                             type: string
+                          externalCertificate:
+                            description: |-
+                              externalCertificate provides certificate contents as a secret reference.
+                              This should be a single serving certificate, not a certificate
+                              chain. Do not include a CA certificate. The secret referenced should
+                              be present in the same namespace as that of the Route.
+                              Forbidden when `certificate` is set.
+                            properties:
+                              name:
+                                description: |-
+                                  name of the referent.
+                                  More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+                                type: string
+                            type: object
+                            x-kubernetes-map-type: atomic
                           insecureEdgeTerminationPolicy:
                             description: |-
                               insecureEdgeTerminationPolicy indicates the desired behavior for insecure connections to a route. While
                               each router may make its own decisions on which ports to expose, this is normally port 80.
 
 
-                              * Allow - traffic is sent to the server on the insecure port (default)
-                              * Disable - no traffic is allowed on the insecure port.
```

Comment:
If customer's had used `Disable` would that be breaking now ?

Reply:
Good catch! This would have been a breaking change if the `Disable` option had been working as expected. If I set the `Disable` option(even without this change) in `ArgoCD` CR the reconciliation fails as the operator is not able to create/update the route on cluster.
```bash
2024-11-04T04:52:24Z INFO controller_argocd Using reencrypt termination policy for the Server Route
2024-11-04T04:52:24Z ERROR Reconciler error {"controller": "argocd", "controllerGroup": "argoproj.io", "controllerKind": "ArgoCD", "ArgoCD": {"name":"openshift-gitops","namespace":"openshift-gitops"}, "namespace": "openshift-gitops", "name": "openshift-gitops", "reconcileID": "06d5311a-3dbd-42e2-9bb8-f46883766ec4", "error": "Route.route.openshift.io \"openshift-gitops-server\" is invalid: spec.tls.insecureEdgeTerminationPolicy: Invalid value: \"Disable\": invalid value for InsecureEdgeTerminationPolicy option, acceptable values are None, Allow, Redirect, or empty"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
/remote-source/deps/gomod/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.2/pkg/internal/controller/controller.go:329
```
Seems like `Disable` was renamed to `None` in OCP 4.12(that is the lowest version I tested with) and we never really update the Openshift api schemas in our source, therefore the `Disable` option will never work for any users and result into error if set in `ArgoCD` CR. 
Since it was never reported to us as a bug, should we assume none of the users are using it and skip documenting the breaking change or you feel, we should?

Reply:
Thanks for the clarification. Its probably good to mention it in the RN.